File: problemas/1000/1051.cs
using System;
using System.Collections.Generic;

namespace uri1050
{
    internal static class Program
    {
        private static void Main()
        {
            double salario = double.Parse(Console.ReadLine());
            double imposto = 0.0;

            var faixas = new (double Valor, double Aliquota)[]
            {
                (4500.00, 0.28),
                (3000.00, 0.18),
                (2000.00, 0.08)
            };

            foreach (var faixa in faixas)
            {
                if (salario > faixa.Valor)
                {
                    double diferenca = salario - faixa.Valor;
                    imposto += diferenca * faixa.Aliquota;
                    salario -= diferenca;
                }
            }

            Console.WriteLine(imposto == 0.0 ? "Isento" : $"R$ {imposto:F2}");
        }
    }
}

File: problemas/1000/1051.php
<?php

$salario = readline();
$imposto = 0.0;

$faixas = array(
    4500.00 => 0.28,
    3000.00 => 0.18,
    2000.00 => 0.08
);

foreach($faixas as $valor => $aliquota) {
    if ($salario > $valor) {
        $diferenca = $salario - $valor;
        $imposto += $diferenca * $aliquota;
        $salario -= $diferenca;
    }
}

echo $imposto == 0.0 ? 'Isento' : 'R$ ' . number_format($imposto, 2, '.', '');
echo "\n";